WORKFLOW.rst

```diff
@@ -0,0 +1,153 @@
+====================================
+ RPC OpenStack Development Workflow
+====================================
+
+Summary
+=======
+
+It is currently necessary, to change the way the RPC OpenStack team develops
+the project (i.e., adds features, fixes bugs). The old workflow is described
+in the :ref:`Old Workflow <old-workflow>` section.
+
+The new workflow is outlined under :ref:`Proposed Workflow
+<proposed-workflow>`
+
+.. _old-workflow:
+
+RPC OpenStack's Old Workflow
+============================
+
+Facts
+-----
+
+#. |rpcos| is configured with the following branches:
+
+   - ``master``
+
+   - ``kilo``
+
+   When we decide to support liberty, a third branch will appear - ``liberty``.
+
+#. Bug fixes move from ``master`` to the other branch(es) via cherry-picks.
+
+#. Bugs are hand-managed by developers and backports are tracked haphazardly.
+
+#. Developers cannot use GitHub's commit message syntax to auto-close bugs.
+
+#. ``master`` 's current focus is not well defined. (Is it the next minor
+   release of the most recent stable branch? Is it the next major release of
+   RPC OpenStack?)
+
+Workflow
+--------
+
+#. A bug is filed or a feature is proposed to |rpcos|
+
+#. After triage and prioritization, a developer creates a fix.
+
+#. The fix is pushed to a branch on that developer's fork.
+
+   This prevents another developer from being able to collaborate on that
+   change.
+
+#. The developer submits a pull request against ``master``. It's reviewed,
+   iterated upon, and eventually merged.
+
+#. If that PR is elligible to be backported, then the associated bug is
+   labeled as such and PRs are submitted to the stable branches.
+
+#. Stable PRs are then reviewed and merged.
+
+#. A developer will (eventually) close the original issue.
+
+.. _proposed-workflow:
+
+RPC OpenStack's Proposed New Workflow
+=====================================
+
+Pre-Workflow
+------------
+
+#. Rename ``master`` to be |lsbnext|.
+
+#. Make |lsbnext| the "default" branch on GitHub.
+
+   This assures that all PRs will be targetted to |lsbnext| by default.
+
+Workflow
+--------
+
+#. After a bug or feature is prioritized and assigned, the developer pushes a
+   fix to a branch on |rpcos|. [#]_
+
+#. A PR is sent to target the |lsbnext| branch (if the bug applies to
+   |lsbnext|, otherwise it is sent to the most recent branch it affects.
```

Comment:
So now that I've added something, I think the workflow I'd like to propose (that's closest to what we have) is something along the lines of the following:

Currently we're still developing things for the 11.x series (kilo). Master would then be named `kilo-next`. When we decide to start doing liberty development, `kilo-next` would become `liberty-next`. When we create r12.0.0 is when we would create a `liberty` branch. That said, if we change our workflow to the strawman below, we would then simply cut `r12.0` branch from `liberty-next` instead.